metaverse_modules/nameplate/index.js

```diff
@@ -0,0 +1,125 @@
+import * as THREE from 'three';
+import metaversefile from 'metaversefile';
+import {GLTFLoader} from 'three/examples/jsm/loaders/GLTFLoader.js';
+import {getModelGeoMat} from './model';
+import {Matrix4} from 'three';
+
+const {useApp, useCamera, useLocalPlayer, useMaterials, useFrame, useText} =
+  metaversefile;
+
+const Text = useText();
+const gltfLoader = new GLTFLoader();
+let nameplateMesh = null;
+
+async function createNameplateMesh() {
+  if (nameplateMesh) return;
```

Comment:
There are a lot of guard conditions like this. Can we program this without all of these checks and just call things in the correct order?

Reply:
I used such conditions to implement the correct MultiMeshInstance function. And separating the features is good for programming, I think. Please let me know if any other solutions.

Reply:
Lets assume single mesh for this case, we will require all future nameplates to be single mesh, single material